### PR TITLE
Mystery crm

### PR DESCRIPTION
**Pull request by junji-genesys**

### CODE REVIEW

File: app/adapters/activity.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                data = {activities:[]};
+            }
+            resolve(data);
+        });   
+    },
+    updateRecord: function(store, type, snapshot){
+        let currentActivities = JSON.parse(window.localStorage.getItem('activities'));
+        let record = store.serializerFor(type.modelName).serialize(snapshot);
+        record.id = snapshot.record.id;
+        if(currentActivities && currentActivities.activities && currentActivities.activities){
```
Comment (mbowergenesys) [inline app/adapters/activity.js]: Is "currentActivities.activities && currentActivities.activities" intentional?

Comment (junji-genesys) [reply app/adapters/activity.js]: I don't think so... I'll remove the duplicate checks.

File: app/services/issue-service.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import Service from '@ember/service';
+import Ember from 'ember';
+
+export default Service.extend({
+    store: Ember.inject.service(),
+    init: function(){
+        this._super(...arguments);
+        this.getIssues().then((data)=>{
+            if(data.get('length') == 0){
+                // console.log('Pre-populating date if none is found');
```
Comment (travis-cawthorn) [inline app/services/issue-service.js]: This should probably be deleted or at-least un commented.

File: app/services/issue-service.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                this.get('store').createRecord('issue', {title:'Haunted Air Filed', description:'A mysterious aircraft appears and lands behind some trees, and out of it comes a mysterious ghost which releases a maniacal laugh.', createdDate: Date.now(), status: 'Open'}).save();
+                this.get('store').createRecord('issue', {title:'When does the theme park open?', description:'I anticipate the opening of a theme park while clam collecting on a nearby beach. But when the theme parks lights come on and the rides start running, when the park is supposed to be closed, I investigated it but was ran off by a strange man who looks like a walking torpedo.', createdDate: Date.now(), status: 'Open'}).save();
+               this.get('store').createRecord('issue', {title:'Lost in a desert gold mine: Help!', description:'I was exploring the desert when I came across an abandon gold mine. I looked inside but quickly got turned around. Now idk if I’m hallucinating but I think I am using a ghastly miner. ……. HELP!', createdDate: Date.now(), status: 'Open'}).save();
```
Comment (travis-cawthorn) [inline app/services/issue-service.js]: indention is off to.

File: app/services/user-service.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import Service from '@ember/service';
+import Ember from 'ember';
+
+export default Service.extend({
+    store: Ember.inject.service(),
+    init: function(){
+        this._super(...arguments);
+        this.getUsers().then((data)=>{
+            if(data.get('length') == 0){
+                // console.log('Pre-populating date if none is found');
```
Comment (travis-cawthorn) [inline app/services/user-service.js]: Same as above, un comment or remove.

File: app/templates/application.hbs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{{top-bar title='Mystery Inc.'}}
+<div class='main'>
+{{global-sidebar}}
```
Comment (travis-cawthorn) [inline app/templates/application.hbs]: indention is off for this block

File: app/templates/issues/view.hbs [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    <div class='attribute'>
+        <span class='label'>Created Date:</span> <span>{{date}}</span>
+    </div>
+    <div class='attribute'>
+        <span class='label'>Status:</span> <span>{{model.status}}</span>
+    </div>
+    <div class='attribute'>
+        <span class='label'>Description:</span> <span>{{model.description}}</span>
+    </div>
+
```
Comment (travis-cawthorn) [inline app/templates/issues/view.hbs]: extra space

File: public/framework.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                window.PureCloud.addCustomAttributes(message.data);
+            } else if(message.type == "processCallLogCompleted"){
+                var mapping = callLogMapping[message.interaction.id];
+                delete callLogMapping[message.interaction.id];
+                if(mapping && mapping.onSuccess){
+                    mapping.onSuccess({
+                        id:message.callLogId
+                    });
+                }
+                // console.log('###### payload', message);
```
Comment (travis-cawthorn) [inline public/framework.js]: same as above either uncomment or remove.

File: app/templates/issues/view.hbs (hunk 6 above)
Comment (junji-genesys) [reply app/templates/issues/view.hbs]: Which space are you talking about? Is it a whitespace between "Description:" and model.description value?

**Pull request merged**